### ISSUE TITLE
Unable to edit job application deadline for some applications

### ISSUE THREAD

**Issue by cyaoxuan** [cyaoxuan/pe#6]
Tried running the command `edit --application 1 --by 10-10-2023` with valid data as seen below, but I get the error that "Job Application with same name found. Set a different name" when the name was not set. For context, I am using the sample data with 1 job application added. I am unsure of what the issue is and how to rectify this issue as a user. This also applies when I edit index 2.

![image.png](https://raw.githubusercontent.com/cyaoxuan/pe/main/files/196a49bd-fc7e-4604-9688-a2d43d47de5b.png)

However, `edit --application 3 --by 10-10-2023` with the same data as above works

![image.png](https://raw.githubusercontent.com/cyaoxuan/pe/main/files/ce76994b-fcca-48ec-b6c6-68a7e25c4ddc.png)

**Comment by nus-pe-script** [comment]
# Team's Response
Unfortunately I am not able to reproduce this bug after multiple tries.

Based on how I am not able to reproduce the bug, I would assume that this is a rare case.

Based on my understanding of how Jobby works, I would assume that the way for the user to resolve this is to close and open the application. This would be a bit of an inconvenience but not a major one, although this was not documented as we did not expect such a bug to occur. 

Regards, I will put medium since it appears to be a legitimate bug that occurs rarely, but I cannot reproduce it.
# Items for the Tester to Verify
## :question: Issue response

Team chose [`response.CannotReproduce`]

- [x] I disagree

**Reason for disagreement:** I tried it again and realised that my previous step of editing the organisation was also relevant to this bug. I apologise for not making it clear previously due to the lack of time. That being said, I hope this does not affect the dev's accuracy marks.

Steps to reproduce:
- Prerequisites: Have an organisation and a job application for that organisation (I used the sample data)
1. Edit the organisation
    - e.g. `edit 1 --url a.com` 
    - Not sure if it applies for all fields, but I tried with address, contact number and url 
2. Edit the job application tied to the organisation 
    - e.g. `edit --application 1 --by 10-10-2023`
    - The error is raised even if I try to edit the different fields including `description`, `by`, `stage` and `status`.
    - Editing the job application's title is ok. In fact, if I edit the title, then edit the problematic fields above, it gets accepted. However, after closing and opening the application, I get an extra job application with the edited title. 

I believe this is due to some kind of mismanagement of the link between the organisation and job application as it is only reproducible when the organisation has been edited. Additionally, there seems to be some issue with the checking of duplicate job applications, hence it is ok to edit by title though it gives rise to an extra job application (might be an outcome of this bug or a separate issue all together, so probably not highly relevant to this bug report).

<catcher-end-of-segment><hr>
## :question: Issue severity

Team chose [`severity.Medium`]
Originally [`severity.High`]

- [x] I disagree

**Reason for disagreement:** As explained in the corrected steps to reproduce, it does not seem very unlikely for such a case to happen as it only requires the editing of the organisation followed by editing the job application. A realistic case I can think of where a user would need to do it is if the point of contact changes between different stages of the interview process (e.g. HR team transfers to the SWE team between the resume and interview process), which would require changes in both the organisation and job application entities. 

While it is true that this issue will be resolved when I close and reopen the app, it did not occur to me as a user due to my lack of understanding of the app, though a user would eventually try that anyways. Nonetheless, I think this bug surfaces a high severity flaw in the application while it is running, specifically regarding the mismanagement of the link between the organisation and job application, a key aspect of the Jobby application.

<catcher-end-of-segment><hr>
